test/key_management.test.js

```diff
@@ -0,0 +1,43 @@
+const web3 = require('web3');
+const { NearProvider, nearWeb3Extensions, nearAPI, utils } = require('../src/index');
+
+const NEAR_ENV = process.env.NEAR_ENV || 'local';
+const config = require('./config')[NEAR_ENV];
+const ACCOUNT = require(config.keyPath);
+const ACCOUNT_ID = ACCOUNT.account_id;
+const ACCOUNT_KEY = ACCOUNT.secret_key;
+const ACCOUNT_KEYPAIR = nearAPI.utils.KeyPair.fromString(ACCOUNT_KEY);
+
+const withWeb3 = (fn) => {
+    const web = new web3();
+    const keyStore = new nearAPI.keyStores.InMemoryKeyStore();
+    keyStore.setKey(NEAR_ENV, ACCOUNT_ID, ACCOUNT_KEYPAIR);
+    let provider = new NearProvider({
+        nodeUrl: config.nodeUrl,
+        keyStore,
+        masterAccountId: ACCOUNT_ID,
+        networkId: NEAR_ENV,
+    });
+    web.setProvider(provider);
+    return () => fn(web);
+};
+
+describe('\n---- KEY MANAGEMENT ----', () => {
+    test('signTypedData', withWeb3(async (web3) => {
+        params = [utils.nearAccountToEvmAddress(ACCOUNT_ID), JSON.parse('{"types":{"EIP712Domain":[{"name":"name","type":"string"},{"name":"version","type":"string"},{"name":"chainId","type":"uint256"},{"name":"verifyingContract","type":"address"}],"AddUserRequest":[{"name":"handle","type":"bytes16"},{"name":"nonce","type":"bytes32"}]},"domain":{"name":"User Factory","version":"1","chainId":"98","verifyingContract":"0x805B9c39D919Dec9eCfF8c10C91479B4d68dA2F8"},"primaryType":"AddUserRequest","message":{"handle":"0x7573657248616e646c6531","nonce":"0xb28a876f70ba058235f107c1572570125abe7201231e2a80217010ea39eab209"}}')];
+        let sig = await new Promise((resolve, reject) => web3.currentProvider.sendAsync({
+            method: 'eth_signTypedData',
+            params,
+            from: utils.nearAccountToEvmAddress(ACCOUNT_ID)
+        }, function (err, result) {
+            if (err) {
+                reject(err)
+            } else if (result.error) {
+                reject(result.error)
+            } else {
+                resolve(result.result)
+            }
+        }));
+        expect(sig).toEqual('0x84af5d73797e9927ac16f3cebb6b6d8fbf90d3c12a876494ef2cb9f3ef6c193e380e01939e76b7ee22fe252cff7afa357151886df4159e360323b354ba95af08');
```

Comment:
We're getting to a good place with this branch, and I'm working on the failing tests. I've checked out the commit (8f061e54e80a73a6346e829f54ae4adb5402ba5f) where this was entered and see it was never passing. I hate to just change this value to what the tests say is the correct signature, but I think that's on the table as this one slipped by my review. If there's reason to think this test is supposed to pass, please let me know.

Reply:
Figured it out and changes a few tests, resolving.